OP2Helper.cpp

```diff
@@ -38,6 +46,89 @@ void InitPlayerResources(int playerNum, const ResourceSet& resourceSet)
 }
 
 
+void CreateTubeLine(LOCATION loc1, LOCATION loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[](LOCATION& location) { TethysGame::CreateWallOrTube(location.x, location.y, 0, map_id::mapTube); }
+	);
+}
+
+void CreateWallLine(LOCATION loc1, LOCATION loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[](LOCATION& location) { TethysGame::CreateWallOrTube(location.x, location.y, 0, map_id::mapWall); }
+	);
+}
+
+void CreateLavaWallLine(LOCATION loc1, LOCATION loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[](LOCATION& location) { TethysGame::CreateWallOrTube(location.x, location.y, 0, map_id::mapLavaWall); }
+	);
+}
+
+void CreateMicrobeWallLine(LOCATION loc1, LOCATION loc2)
+{
+	ExecuteAcrossLine(loc1, loc2, 
+		[](LOCATION& location) { TethysGame::CreateWallOrTube(location.x, location.y, 0, map_id::mapMicrobeWall); }
+	);
+}
+
+void RecordTubeLine(BuildingGroup& buildingGroup, LOCATION loc1, LOCATION loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[ &buildingGroup ] (LOCATION& location) { buildingGroup.RecordTube(location); }
+	);
+}
+
+void RecordWallLine(BuildingGroup& buildingGroup, const LOCATION& loc1, const LOCATION& loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[ &buildingGroup ] (LOCATION& location) { buildingGroup.RecordWall(location, map_id::mapWall); }
+	);
+}
+
+void RecordLavaWallLine(BuildingGroup& buildingGroup, const LOCATION& loc1, const LOCATION& loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[ &buildingGroup ] (LOCATION& location) { buildingGroup.RecordWall(location, map_id::mapLavaWall); }
+	);
+}
+
+void RecordMicrobeWallLine(BuildingGroup& buildingGroup, const LOCATION& loc1, const LOCATION& loc2)
+{
+	ExecuteAcrossLine(loc1, loc2,
+		[ &buildingGroup ] (LOCATION& location) { buildingGroup.RecordWall(location, map_id::mapMicrobeWall); }
+	);
+}
+
+void ExecuteAcrossLine(LOCATION loc1, LOCATION loc2, const std::function <void(LOCATION&)>& recordFunction)

```

Comment:
Happy to entertain different name for function and passed in lambda. Don't want to spend forever trying to find perfect name either though.

Reply:
I would name it just `function`. The function is being used for both Create and Record, so probably best to drop the "record" part as it is too specific.

Reply:
I meant record in the sense of generally record an event at each location in the L. However, it is confusing sense there is a record wall/tube as well. I think function as the name will work fine. I assumed it was a keyword for some reason, but looks like you added it fine.